addons/project/views/project_views.xml

```diff
@@ -58,6 +70,7 @@
                         <filter string="Personal Stage" name="personal_stage" context="{'group_by': 'personal_stage_type_ids'}"/>
                         <filter string="Assignees" name="user" context="{'group_by': 'user_ids'}"/>
                         <filter string="Project" name="project" context="{'group_by': 'project_id'}"/>
+                        <filter string="Parent Task" name="parent_task" context="{'group_by': 'parent_id'}" groups="project.group_subtask_project"/>
```

Comment:
Why not using `Ancestor Task` and its related field `ancestor_id` ? I do not see the point of searching a task by its direct parent. As this information is already present in the form of the parent task. Could you challenge @aju-odoo on this? Thanks 🙂